src/Cameras/Inputs/BaseCameraMouseWheelInput.ts

```diff
@@ -0,0 +1,202 @@
+                this._wheelDeltaY -=
+                    this.wheelPrecisionY * platformScale *
+                    (<any>event).wheelDeltaY / this._normalize;
+                this._wheelDeltaZ +=
+                    this.wheelPrecisionZ * platformScale *
+                    (<any>event).wheelDeltaZ / this._normalize;
+            } else if ((<any>event).wheelDelta) {
+                // IE >= v9   (Has WebGL >= v11)
+                // Maybe others?
+                this._wheelDeltaY -=
+                    this.wheelPrecisionY * (<any>event).wheelDelta / this._normalize;
+            } else if (event.detail) {
+                // Firefox < v17  (Has WebGL >= v4)
+                // TODO How should we scale this?
+                // Since it's Firefox, it's probably the same as
+                // WheelEvent.DOM_DELTA_LINE.
+                // ie: we can presume it needs scaled to match per-pixel.
+                this._wheelDeltaY +=
+                    this.wheelPrecisionY * this._ffMultiplier * event.detail;
+                if ("axis" in event &&
+                      (<any>event).axis === (<any>event).HORIZONTAL_AXIS) {
+                    this._wheelDeltaX = this._wheelDeltaY;
+                    this._wheelDeltaY = 0;
+                }
+            }
+
+            if (event.preventDefault) {
+                if (!noPreventDefault) {
+                    event.preventDefault();
+                }
+            }
+        };
+
+        this._observer = this.camera.getScene().onPointerObservable.add(
+            this._wheel,
+            PointerEventTypes.POINTERWHEEL);
+    }
+
+    /**
+     * Detach the current controls from the specified dom element.
+     * @param element Defines the element to stop listening the inputs from
+     */
+    public detachControl(element: Nullable<HTMLElement>): void {
+        if (this._observer && element) {
+            this.camera.getScene().onPointerObservable.remove(this._observer);
+            this._observer = null;
+            this._wheel = null;
+        }
+        if (this.onChangedObservable) {
+            this.onChangedObservable.clear();
+        }
+    }
+
+    /**
+     * Called for each rendered frame.
+     */
+    public checkInputs(): void {
+        this.onChangedObservable.notifyObservers({
+            wheelDeltaX: this._wheelDeltaX,
+            wheelDeltaY: this._wheelDeltaY,
+            wheelDeltaZ: this._wheelDeltaZ
+        });
+
+        // Clear deltas.
+        this._wheelDeltaX = 0;
+        this._wheelDeltaY = 0;
+        this._wheelDeltaZ = 0;
+    }
+
+    /**
+     * Gets the class name of the current intput.
+     * @returns the class name
+     */
+    public getClassName(): string {
+        return "BaseCameraMouseWheelInput";
+    }
+
+    /**
+     * Get the friendly name associated with the input class.
+     * @returns the input friendly name
+     */
+    public getSimpleName(): string {
+        return "mousewheel";

```

Comment:
Just out of curiosity, what scenarios would this be used in?  Also, does this have any relation to the string used for mouse wheel events?

Reply:
This is part of the input loading infrastructure.
If you look at the contents of `camera.inputs.attached` in any playground, this value is the key in that collection relating to the `ICameraInput` instance.
See https://doc.babylonjs.com/how_to/customizing_camera_inputs for more detail.